2112 feat button design guidelines

kathydlavoie-ps opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Closes #2112

## What is the new behavior?

* Reorganizes docs into individual pages after consulting with @bradryanbice 
* Includes content for `Overview` and `Usage` pages

## Other information

https://github.com/pluralsight/pando/assets/146388897/b5c9ac71-6970-4dc0-b402-d744cd4d0afd

bradryanbice reviewed (dismissed):
Looks good, a few observations:

* I think it would be ideal for the side nav "Button" link to go directly to the Overview page, if possible
* Do we need or want to have the individual nested pages for each component in the side nav? It might be overkill, especially if each component has all 4 links.
* I'm wondering how we can make the Overview page more useful, so that it's not always an interruptive experience if you're usually trying to get to implementation or usage docs. Or is it just standard? Perhaps a question for testing to solve for us.

kathydlavoie-ps commented:
After talking with @bradryanbice , merging with these changes:

* removed page links from main left nav (button nav item goes to button overview)
* Internal Button Pages are navigated to from bottom nav within pages
* moved Anatomy to Usage page
* Included a dev "overview" on front page summarizing props & including a live example, design overview tbd
